### PR TITLE
feat(dapps) check Wallet Connect pair uri for already used and expired

### PR DESCRIPTION
**Pull request by stefandunca**
### Updates: #14676

https://github.com/status-im/status-desktop/assets/47554641/0e6ac19e-b292-454c-b3c1-d6696fd259d2

### CODE REVIEW

File: ui/app/AppLayouts/Wallet/panels/DAppsWorkflow.qml [hunk 1]
```diff
@@ -151,8 +155,10 @@ DappsComboBox {
                 function onMaxFeesUpdated(maxFees, maxFeesWei, haveEnoughFunds, symbol) {
                     maxFeesText = `${maxFees.toFixed(2)} ${symbol}`
```
Comment (caybro) [inline ui/app/AppLayouts/Wallet/panels/DAppsWorkflow.qml:156]: This should be formatted using `LocaleUtils`; kinda hard to do with the `ethStr` below... :thinking:

**Review by caybro — APPROVED** [review]
LGTM otherwise

File: ui/app/AppLayouts/Wallet/panels/DAppsWorkflow.qml (hunk 1 above)
Comment (stefandunca) [reply ui/app/AppLayouts/Wallet/panels/DAppsWorkflow.qml:156]: This is just a dummy display code to show that the data is coming from the SDK. It has to be replaced with the proper status-go based implementation here in this work package: https://github.com/status-im/status-desktop/issues/15192

**Comment by alexjba** [comment]
Now I see we're missing the flow where if the url is valid we should automatically move forward and open the connect dialog. Do you mind adding this here? Or we can create another task so that we don't forget about it.

File: ui/imports/shared/popups/walletconnect/PairWCModal/WCUriInput.qml
Comment (alexjba) [inline ui/imports/shared/popups/walletconnect/PairWCModal/WCUriInput.qml:1]: WDYT of keeping all the validation rules in a single place?
Now it's split between `WCUriInput` and `WalletConnectService`

Comment (alexjba) [inline ui/imports/shared/popups/walletconnect/PairWCModal/WCUriInput.qml:1]: I think now we can remove this
```
        function wcUriAlreadyUsed(uri) {
            // TODO: Check if URI is already used
            return false
        }
```

File: ui/app/AppLayouts/Wallet/panels/DAppsWorkflow.qml (hunk 1 above)
Comment (caybro) [reply ui/app/AppLayouts/Wallet/panels/DAppsWorkflow.qml:156]: Ah cool

**Review by alexjba — APPROVED** [review]
Nice! There is some code we need to remove here.
Other than that, LGTM!

**Comment by alexjba** [comment]
qml tests tail with `SwapInputPanel ` error. I've thought we fixed this. Is this an old branch?
```
[2024-07-07T13:16:55.175Z] 3: FAIL!  : QmlTests::SwapInputPanel::test_input_greater_than_max_balance() Compared values are not the same
[2024-07-07T13:16:55.175Z] 3:    Actual   (): 0
[2024-07-07T13:16:55.175Z] 3:    Expected (): 5.42
[2024-07-07T13:16:55.175Z] 3:    Loc: [/home/jenkins/workspace/s_linux_x86_64_tests-ui_PR-15468/storybook/qmlTests/tests/tst_SwapInputPanel.qml(424)]
```

**Comment by caybro** [comment]
> qml tests tail with `SwapInputPanel ` error. I've thought we fixed this. Is this an old branch?
> 
> ```
> [2024-07-07T13:16:55.175Z] 3: FAIL!  : QmlTests::SwapInputPanel::test_input_greater_than_max_balance() Compared values are not the same
> [2024-07-07T13:16:55.175Z] 3:    Actual   (): 0
> [2024-07-07T13:16:55.175Z] 3:    Expected (): 5.42
> [2024-07-07T13:16:55.175Z] 3:    Loc: [/home/jenkins/workspace/s_linux_x86_64_tests-ui_PR-15468/storybook/qmlTests/tests/tst_SwapInputPanel.qml(424)]
> ```

Flaky test, I've just seen it too, restart helped

**Comment by stefandunca** [comment]
Done, Thanks for the great input

> Now I see we're missing the flow where if the url is valid we should automatically move forward and open the connect dialog. Do you mind adding this here? Or we can create another task so that we don't forget about it.

I implemented and tested it within this PR

> WDYT of keeping all the validation rules in a single place?

I moved them all in Service, maybe we could even get further in a follow up change and group all the pairing logic under the sme file/object.

**Pull request merged**